### PR TITLE
Fix setting focus on RuiButton and RuiExpandCollapse

### PR DESCRIPTION
**Pull request by ricominten**
using `focus()` now works for both RuiButton and RuiExpandCollapse

### CODE REVIEW

File: components/RuiButton/src/RuiButton.ts [hunk 1]
```diff
@@ -83,6 +89,20 @@ export class RuiButton extends LitElement {
 
 	/* #region Methods */
```
Comment (jennasalau) [inline components/RuiButton/src/RuiButton.ts]: Can you also please add

```
protected createRenderRoot() {
		return this.attachShadow({mode: 'open', delegatesFocus: true});
}
```

File: components/RuiExpandCollapse/src/RuiExpandCollapse.ts [hunk 2]
```diff
@@ -189,6 +200,13 @@ export class RuiExpandCollapse extends LitElement {
 		}
 	}
```
Comment (jennasalau) [inline components/RuiExpandCollapse/src/RuiExpandCollapse.ts]: Can you please also add

```
protected createRenderRoot() {
		return this.attachShadow({mode: 'open', delegatesFocus: true});
}
```

File: components/RuiExpandCollapse/src/RuiExpandCollapse.ts [hunk 3]
```diff
@@ -18,6 +18,23 @@ type RuiExpandCollapsePropertyType = boolean;
...
  */
 export class RuiExpandCollapse extends LitElement {
+
+	/**
+	 * Internal open state of component
+	 */
+	private _open: boolean = false;
+	private _collapseableEl: HTMLDivElement | null = null;
+	private _detailsSlotEl: HTMLSlotElement | null = null;
+	private _uuid: string = this._generateUUIDv4();
```
[the remark this answers is not in the document]
Comment (ricominten) [reply components/RuiExpandCollapse/src/RuiExpandCollapse.ts]: fair point. This was inherited so will remove

File: components/RuiButton/src/RuiButton.ts (hunk 1 above)
Comment (ricominten) [reply components/RuiButton/src/RuiButton.ts]: yep can do. that said delegatesFocus is very poorly supported right now.

File: components/RuiButton/src/RuiButton.ts [hunk 4]
```diff
@@ -83,6 +89,20 @@ export class RuiButton extends LitElement {
...
 
+	/**
+	 * LIT firstUpdated lifecycle event
+	 */
+	public firstUpdated():void {
```
[the remark this answers is not in the document]
Comment (ricominten) [reply components/RuiButton/src/RuiButton.ts]: Yeah I'm having a poor experience with the `connectedCallback`, and given that the shadow doesn't change it's safe to rely on just this for this purpose (it's the same as componentDidMount in React.

**Pull request merged**